tests/php/Model/RedirectorPageTest.php

```diff
@@ -117,4 +124,25 @@ public function testNoJSLinksAllowed()
         $page->write();
         $this->assertEmpty($page->ExternalURL);
     }
+
+    public function testOnMissingLinkResponse()
+    {
+        $page = $this->objFromFixture(RedirectorPage::class, 'goodinternal');
+        $this->assertEquals('/redirection-dest/', $page->Link());
+        $response = $this->get($page->Link());
+        $this->assertEquals(200, $response->getStatusCode());
+
+        // Override default response code to 400
+        Config::forClass(RedirectorPageController::class)->set('response_code', 404);
+
+        // Unpublish the target page of this redirector page.
+        $targetPage = $this->objFromFixture(Page::class, 'dest');
```

Comment:
there is no test here for setting a response code and having a published page that responds successfully.